filebeat/publish/publish.go

```diff
@@ -172,19 +172,28 @@ func (p *asyncLogPublisher) Start() {
 			case <-p.done:
 				return
 			case events := <-p.in:
-				pubEvents := make([]common.MapStr, len(events))
-				for i, event := range events {
+
+				tmpEvents := make([]*input.Event, 0)
+				for _, event := range events {
+					if event.HasData() {
+						tmpEvents = append(tmpEvents, event)
+					}
+				}
```

Comment:
No need to copy. I was thinking that eventsBatch also needs the filtered list, but if it is for the registrar this is not the case. 

I simplified the implementation. Please have a look again.

The sync publisher also does the filtering on line 114.